zuul-core/src/main/java/com/netflix/zuul/filters/common/GZipResponseFilter.java

```diff
@@ -78,7 +78,7 @@ public boolean shouldFilter(HttpResponseMessage response) {
 
         // Check the headers to see if response is already gzipped.
         final Headers respHeaders = response.getHeaders();
-        boolean isResponseGzipped = HttpUtils.isGzipped(respHeaders);
+        boolean isResponseGzipped = HttpUtils.isGzipped(respHeaders) || HttpUtils.isDeflated(respHeaders);
```

Comment:
Minor nitpick: rename the variable to isResponseCompressed since we are checking for 2 different compression types.